Sources/tuist/tuist.docc/Articles/Tuist/Users/synthesized-files.md

```diff
@@ -0,0 +1,95 @@
+
+Tuist solves the problems above by **synthesizing a unified interface to access bundles and resources** that abstracts away the implementation details.
+
+> Important: Even though accessing resources through the Tuist-synthesized interface is not mandatory, we recommend it because it makes the code easier to reason about and the resources to move around.
+
+### Synthesized resources
+
+Tuist provides interfaces to declare the content of files such as `Info.plist` or entitlements in Swift.
+This is useful to ensure consistency across targets and projects,
+and leverage the compiler to catch issues at compile time.
+You can also come up with your own abstractions to model the content and share it across targets and projects.
+
+When your project is generated,
+Tuist will synthesize the content of those files and write them into the `Derived` directory relative to the directory containing the project that defines them.
+
+> Note: We recommend adding the `Derived` directory to the `.gitignore` file of your project.
+
+### Synthesized bundle accessor
+
+Tuist synthesizes an interface to access the bundle that contains the target resources.
+
+##### Swift
+
+The target will contain an extension of the `Bundle` type that exposes the bundle:
+
+```swift
+let bundle = Bundle.module
+```
+
+##### Objective-C
+
+In Objective-C, you'll get an interface `{Target}Resources` to access the bundle:
+
+```objc
+NSBundle *bundle = [MyFeatureResources bundle];
+```
+
+> Tip: If a target product, for example a library, doesn't support resources, Tuist will include the resources in a target of product type `bundle` ensuring that it ends up in the final product and that the interface points to the right bundle.
+
+### Synthesized resource accessors
+
+Resources are identified by their name and extension using strings. This is not ideal because it's not caught at compile time and might lead to crashes in release. To prevent that, Tuist integrates [SwiftGen](https://github.com/SwiftGen/SwiftGen) into the project generation process to synthesize an interface to access the resources. Thanks to that, you can confidently access the resources leveraging the compiler to catch any issues.
+
+Tuist includes [templates](https://github.com/tuist/tuist/tree/main/Sources/TuistGenerator/Templates) to synthesize accessors for the following resource types by default:
+
+| Resource type | Synthesized file |
+| --- | ---- |
+| Images and colors | `Assets+{Target}.swift` |
+| Strings | `Strings+{Target}.swift` |
+| Plists | `{NameOfPlist}.swift` |
+| Fonts | `Fonts+{Target}.swift` |
+| Files | `Files+{Target}.swift` |
+
+> Note: You can disable the synthesizing of resource accessors in a per-project basis by passing the `disableSynthesizedResourceAccessors` option to the project options.
+
+#### Custom templates
```

Comment:
Do we have any fixture showcasing this? Maybe we could link it here?

Reply:
I included a reference at the bottom.